coriolis/tests/conductor/rpc/test_server.py

```diff
@@ -18,6 +22,345 @@ def setUp(self):
+        mock_get_service.assert_called_once_with(
+            mock.sentinel.context, mock.sentinel.worker_id)
+
+        self.assertEqual(result, mock_from_service_definition.return_value)
+
+    @mock.patch.object(server.ConductorServerEndpoint, 'get_endpoint')
+    @mock.patch.object(db_api, 'delete_endpoint')
+    @mock.patch.object(db_api, 'update_endpoint')
+    @mock.patch.object(db_api, 'add_endpoint')
+    @mock.patch.object(models, 'Endpoint')
+    @mock.patch.object(uuid, 'uuid4')
+    def test_create_endpoint(self, mock_uuid4, mock_endpoint_model, mock_add_endpoint, mock_update_endpoint, mock_delete_endpoint, mock_get_endpoint):
+        endpoint = self.server.create_endpoint(
+            mock.sentinel.context, mock.sentinel.name, mock.sentinel.endpoint_type, mock.sentinel.description, mock.sentinel.connection_info)
+        self.assertEqual(mock_endpoint_model.return_value.name,
+                         mock.sentinel.name)
+        self.assertEqual(mock_endpoint_model.return_value.type,
+                         mock.sentinel.endpoint_type)
+        self.assertEqual(mock_endpoint_model.return_value.description,
+                         mock.sentinel.description)
+        self.assertEqual(mock_endpoint_model.return_value.connection_info,
+                         mock.sentinel.connection_info)
+
+        mock_add_endpoint.assert_called_once_with(
+            mock.sentinel.context, mock_endpoint_model.return_value)
+        mock_update_endpoint.assert_not_called()
+        mock_delete_endpoint.assert_not_called()
+        self.assertEqual(endpoint, mock_get_endpoint.return_value)
+
+        # mapped_regions exist
+        self.server.create_endpoint(
+            mock.sentinel.context, mock.sentinel.name, mock.sentinel.endpoint_type, mock.sentinel.description, mock.sentinel.connection_info, mock.sentinel.mapped_regions)
+
+        mock_update_endpoint.assert_called_once_with(
+            mock.sentinel.context, str(mock_uuid4.return_value), {'mapped_regions': mock.sentinel.mapped_regions})
+        mock_delete_endpoint.assert_not_called()
+
+        # mapped_regions exist and there's an error updating the endpoint
+        mock_update_endpoint.side_effect = Exception()
+        self.assertRaises(Exception, lambda: self.server.create_endpoint(
+            mock.sentinel.context, mock.sentinel.name, mock.sentinel.endpoint_type, mock.sentinel.description, mock.sentinel.connection_info, mock.sentinel.mapped_regions))
+        mock_delete_endpoint.assert_called_once_with(
+            mock.sentinel.context, str(mock_uuid4.return_value))
+
+    @mock.patch.object(db_api, 'get_endpoint')
+    @mock.patch.object(db_api, 'update_endpoint')
+    def test_update_endpoint(self, mock_update_endpoint, mock_get_endpoint):
+        endpoint = testutils.get_wrapped_function(
+            self.server.update_endpoint)(self, mock.sentinel.context, mock.sentinel.endpoint_id, mock.sentinel.updated_values)  # type: ignore
```

Comment:
You could have added a linebreak before `(<\n>self, mock.sentinel.context, ...)` so the call args on a separate line to satisfy the linter.